hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -488,7 +488,7 @@ public void applySelector(ClientSelector newSelector) {
             String memberUuid = ownershipMappings.get(clientUuid);
             ClientDisconnectionOperation op = new ClientDisconnectionOperation(clientUuid, memberUuid);
             InternalOperationService service = nodeEngine.getOperationService();
-            Address thisAddress = getThisAddress();
+            Address thisAddress = node.getThisAddress();
```

Comment:
Since #14521 `ClientEngine.getThisAddress` returns the client protocol address of the local member. This is the same as member address with single-server-socket network config but different when advanced network config is enabled. Here we want to target operations to this member, so `node.getThisAddress` must be used instead.